### DISCUSSION TITLE
Duplicate headers

### DISCUSSION THREAD

**Discussion by JokerDevops** [fastapi/fastapi#6116] [Questions] [answered]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

Hi,
  I like fastapi very much, but I encountered a problem during my study.

  My code :
```
from typing import List, Union

from fastapi import FastAPI, Header

app = FastAPI()


@app.get("/items/")
async def read_items(x_token: Union[List[str], None] = Header(default=None)):
    return {"X-Token values": x_token}
```

I got a 422 return code not 200
<img width="1483" alt="image" src="https://user-images.githubusercontent.com/41281305/219401923-3c007af4-5f45-40e9-8f89-934073bdaabb.png">

According to the document description, I think I will get the following content:
```
{
    "X-Token values": [
        "bar",
        "foo"
    ]
}
```

**Comment by odiseo0** [comment, marked as answer]
Probably one of your other routes it's messing with it. Your example works for me

**Comment by JokerDevops** [comment]
Thank you for your reminder. I really didn't notice my other routes. It conflicts with other routes. I solved it. Now it is OK
